### PR TITLE
Filters JS refactor

### LINKED ISSUE #554: Refactor filters JS to reduce complexity

### PR DESCRIPTION
**Pull request by steverydz**
## Done

Refactored JS for filters, including removing unused code and renaming functions and variables for improved readability.

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://localhost:8045
- Play with the filters and platform switcher
- Check that it still works as it does on production

## Issue / Card

Fixes #554

### CODE REVIEW

**Review by jkfran — APPROVED** [review]

File: templates/partial/_filters.html [hunk 1]
```diff
@@ -109,14 +109,6 @@ <h3 class="p-card__title p-heading--4 u-no-margin--bottom entity-card-title"></h
   </div>
 </template>
 
-<template id="no-results-message">
-  <div class="p-notification--information">
-    <p class="p-notification__response">
-      We couldn't find anything for your search<span class="search-query"></span>.
```
Comment (anthonydillon) [inline templates/partial/_filters.html:115]: Shouldn't we keep this message?

Comment (steverydz) [reply templates/partial/_filters.html:115]: It's never seen since the search results are now on their own page

Comment (jkfran) [reply templates/partial/_filters.html:115]: I thought about it as well but since we have now a different template for searches we don't need it.

Comment (anthonydillon) [reply templates/partial/_filters.html:115]: Ah of course :+1:

**Pull request merged**